internal/clients/http/azure/azure_client.go

```diff
@@ -69,3 +69,10 @@ func (c *client) Status(ctx context.Context) error {
 	}
 	return nil
 }
+
+func CheckSourceAvailability(ctx context.Context, auth *clients.Authentication, c chan clients.StatusResponse) {
+	c <- clients.StatusResponse{
+		Status:   clients.StatusUnavailable,
+		SourceID: nil,
+	}
+}
```

Comment:
Those channels you want to use to communicate goroutines should be only in the `main` package, you do not want the client API to be aware that they are called like this (from other goroutines).